dhall/src/Dhall/Import/Types.hs

```diff
@@ -38,20 +38,27 @@ import qualified Data.Text
 --   those are well-typed (either of type `List { header : Text, value Text}` or
 --   `List { mapKey : Text, mapValue Text})` and in normal form. These
 --   invariants are preserved by the API exposed by @Dhall.Import@.
-newtype Chained = Chained { chainedImport :: Import }
+newtype Chained = Chained
+    { chainedImport :: Import
+      -- ^ The underlying import
+    }
   deriving (Eq, Ord)
 
 instance Pretty Chained where
     pretty (Chained import_) = pretty import_
 
+-- | An import that has been fully interpeted
 data ImportSemantics = ImportSemantics
-    { importSemantics :: Expr Src X
+    { importSemantics :: Expr Src Void
     -- ^ The fully resolved import, typechecked and beta-normal.
     }
 
 -- | `parent` imports (i.e. depends on) `child`
 data Depends = Depends { parent :: Chained, child :: Chained }
 
+{-| This enables or disables the semantic cache (a faster cache for file-based
+    imports that works even for unprotected imports)
```

Comment:
"Semantic cache" is the name of the cache of protected / hashed imports. But this seems to describe the semi-semantic cache.

Maybe we should find better names…